Translate FAQ file structure doc (#102)
reactjs/de.reactjs.org: content/docs/faq-structure.md

---
id: faq-structure

> title: Datenstruktur

Reviewer: ```suggestion
title: Dateistruktur
```

permalink: docs/faq-structure.html
layout: docs
category: FAQ
---

> ### Gibt es einen empfohlene Datenstruktur für React-Projekte? {#is-there-a-recommended-way-to-structure-react-projects}

Reviewer: ```suggestion
### Gibt es eine empfohlene Dateistruktur für React-Projekte? {#is-there-a-recommended-way-to-structure-react-projects}
```

> React hat keine Meinung dazu, wie Sie Dateien in Ordnern ablegen. Es gibt jedoch einige gängige Ansätze, die Sie in Betracht ziehen sollten.

Reviewer: ```suggestion
React hat keine Meinung dazu, wie du Dateien in Ordnern ablegst. Es gibt jedoch einige gängige Ansätze, die du in Betracht ziehen solltest.
```

Reviewer: We decided to go with "du"

#### Gruppierung nach Merkmalen oder Routen {#grouping-by-features-or-routes}

> Eine gängige Methode zum Strukturieren von Projekten ist CSS, JS und Tests zu finden, die nach Merkmalen oder Routen in Ordnern gruppiert werden können.

Reviewer: ```suggestion
Eine gängige Methode zum Strukturieren von Projekten ist CSS, JS und Tests zu finden, die nach Features oder Routen in Ordnern gruppiert werden können.
```

  ProfileAPI.js
```

> Die Definition eines "Merkmals" ist nicht universell und es liegt an Ihnen, die Granularität zu wählen. Wenn es Ihnen schwer fällt sich eine top-level Ordnerstruktur zu erdenken, können Sie die Benutzer Ihres Produktes fragen aus welchen Hauptbestandteilen es aufgebaut ist, und dieses Modell als Entwurf verwenden.

Reviewer: ```suggestion
Die Definition eines "Features" ist nicht universell und es liegt an dir, die Granularität zu wählen. Wenn es dir schwer fällt eine top-level Ordnerstruktur zu erdenken, kannst du die Benutzer deines Produktes fragen aus welchen Hauptbestandteilen es aufgebaut ist, und dieses Modell als Entwurf verwenden.
```
I think, no one knows what "Merkmale" means in this case.

  ProfileHeader.css
```

> Einige Leute gehen einen Schritt weiter und sortieren Komponenten nach ihrer Rolle in der Anwendung. Das [Atomic Design](http://bradfrost.com/blog/post/atomic-web-design/) ist ein Beispiel was die auf diesem Prinzip basiert ist. Denken Sie daran, dass es oft produktiver ist, solche Methoden als hilfreiche Beispiele, anstatt Regeln zu betrachten.

Reviewer: ```suggestion
Einige Leute gehen einen Schritt weiter und sortieren Komponenten nach ihrer Rolle in der Anwendung. Das [Atomic Design](http://bradfrost.com/blog/post/atomic-web-design/) ist ein Beispiel was auf diesem Prinzip basiert. Denke daran, dass es oft produktiver ist, solche Methoden als hilfreiche Beispiele, anstatt Regeln zu betrachten.
```

> #### Vermeiden Sie zu viel Verschachtelung {#avoid-too-much-nesting}

Reviewer: ```suggestion
#### Vermeide zu viele Verschachtelungen {#avoid-too-much-nesting}
```

> Es gibt viele Probleme wenn JavaScript Projekte zu tief verschachtelt werden. Es wird schwieriger, relative Importe zwischen Dateien zu erstellen oder diese Importe zu aktualisieren, wenn Dateien verschoben werden. Sie sollten sich auf eine drei bis vier Ordner tiefe Struktur beschränken, es sei denn Sie haben einen guten Grund für eine Tiefere. Dies ist natürlich nur eine Empfehlung und möglicherweise für Ihr Projekt nicht relevant.

Reviewer: ```suggestion
Es gibt viele Probleme wenn JavaScript Projekte zu tief verschachtelt werden. Es wird schwieriger, relative Importe zwischen Dateien zu erstellen oder diese Importe zu aktualisieren, wenn Dateien verschoben werden. Du solltest dich innerhalb deines Projektes auf maximal drei bis vier verschachtelte Ordner beschränken, es sei denn du hast einen guten Grund für eine tiefere Ordnerstruktur. Dies ist natürlich nur eine Empfehlung und möglicherweise für dein Projekt nicht relevant.
```

> #### Überdenken Sie es nicht {#dont-overthink-it}

Reviewer: ```suggestion
#### Denke nicht zu viel darüber nach {#dont-overthink-it}
```

> Wenn Sie ein neues Projekt anfangen, [verbringen Sie nicht mehr als fünf Minuten](https://de.wikipedia.org/wiki/Paralyse_durch_Analyse) um eine Datenstruktur auszuwählen. Wählen Sie einen der oben genannten Ansätze (oder überlegen Sie sich einen eigenen) und legen Sie los! Sie werden die Datenstruktur wahrscheinlich überdenken wollen, nachdem Sie ein wenig Code geschrieben haben.

Reviewer: ```suggestion
Wenn du ein neues Projekt anfängst, [verbringe dafür nicht mehr als fünf Minuten](https://de.wikipedia.org/wiki/Paralyse_durch_Analyse) um eine Dateistruktur auszuwählen. Wähle einen der oben genannten Ansätze (oder überlege dir einen eigenen) und leg los! Du wirst die Dateistruktur wahrscheinlich sowieso überdenken, nachdem du ein wenig Code geschrieben hast.
```

> Wenn Sie sich nicht sicher fühlen, sollten Sie zunächst alle Dateien in einem einzigen Ordner speichern. Irgendwann wird dieser so groß werden, dass Sie einige Dateien von dem Rest trennen möchten. Zu diesem Zeitpunkt werden Sie genug Wissen, um festzustellen zu können, welche Dateien Sie am häufigsten gemeinsam bearbeiten. Im Allgemeinen ist es eine gute Idee, Dateien, die Sie häufig gemeinsam ändern, nahe beieinander zu abzulegen. Dieses Prinzip nennt man "Kollokation".

Reviewer: ```suggestion
Wenn du dich nicht sicher fühlst, solltest du zunächst alle Dateien in einem einzigen Ordner speichern. Irgendwann wird dieser so groß werden, dass du einige Dateien von dem Rest trennen möchtest. Zu diesem Zeitpunkt wirst du genug Wissen, um festzustellen zu können, welche Dateien du am häufigsten gemeinsam bearbeitest. Im Allgemeinen ist es eine gute Idee, Dateien, die du häufig gemeinsam änderst, nahe beieinander abzulegen. Dieses Prinzip nennt man "Kollokation (engl. colocation)".
```
